### PR TITLE
[ci] Update pin ci script

### PR DESCRIPTION
**Pull request by gares**
There is also a very urgent fix for the overlay check

### CODE REVIEW

File: dev/tools/pin-ci.sh [hunk 1]
```diff
@@ -38,9 +38,7 @@ process_development() {
 # Execute the script to set the overlay variables
 . $OVERLAYS
 
-# Find all variables declared in the base overlay of the form *_CI_GITURL
-for REPO_VAR in $(compgen -A variable | grep _CI_GITURL)
+for project in ${projects[*]}
```
Comment (SkySkimmer) [inline dev/tools/pin-ci.sh:41]:
```suggestion
for project in ${projects[@]}
```

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
I let you set your preferred milestone

**Comment by gares** [comment]
done thanks

**Comment by SkySkimmer** [comment]
@coqbot: merge now

**Pull request merged**